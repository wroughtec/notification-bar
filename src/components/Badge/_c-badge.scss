.c-badge {
  position: absolute;
  top: 0;
  right: 0;
  z-index: 1;

  display: flex;
  width: 1.5rem;
  height: 1.5rem;

  flex-wrap: wrap;
  justify-content: center;

  align-items: center;
  align-content: center;

  font-size: .75rem;
  color: var(--notificationColour);

  background-color: var(--notificationBackground);
  border-radius: 50%;
}
