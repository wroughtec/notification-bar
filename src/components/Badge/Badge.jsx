// @flow

import React from 'react';
import './_c-badge.scss';

type Props = {
  unreadNotifications: number
};

export const Badge = ({ unreadNotifications }: Props) => {
  if (!unreadNotifications) return null;

  return <span className="c-badge">{unreadNotifications}</span>;
};
