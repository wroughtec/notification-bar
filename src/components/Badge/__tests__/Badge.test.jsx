import fetchMock from 'fetch-mock';
import React from 'react';
import { shallow } from 'enzyme';
import { Badge } from 'components/Badge/Badge';

describe('Badge', () => {
  it('should render correctly with data', () => {
    const props = { unreadNotifications: 6 };
    const component = shallow(<Badge {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should not render without data', () => {
    const props = { unreadNotifications: 0 };
    const component = shallow(<Badge {...props} />);
    expect(component).toMatchSnapshot();
  });
});
