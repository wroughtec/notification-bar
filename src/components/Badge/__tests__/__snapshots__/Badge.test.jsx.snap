// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should not render without data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    unreadNotifications={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Badge should render correctly with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    unreadNotifications={6}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": 6,
      "className": "c-badge",
    },
    "ref": null,
    "rendered": 6,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 6,
        "className": "c-badge",
      },
      "ref": null,
      "rendered": 6,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
